be/src/main/java/com/issuetracker/service/IssueListService.java

```diff
@@ -1,87 +1,68 @@
      * 동일 이슈에 대해서 여러 개의 라벨을 Issue 객체에 list 타입으로 넣어주기 위해서 다소 지저분한 mapping 로직으로 구현했습니다.
      */
-    public IssueListDto fetchMain() {
-        IssueListDto issueListDto = new IssueListDto();
-        List<IssueDao> issueMainPageDtoList = labelRepository.getIssues(true);
+    public IssueListDto fetchMain(Filter filterDto) {
 
-        List<IssueDto> issueDtoList = new ArrayList<>();
-        for (int i = 0; i < issueMainPageDtoList.size(); i++) {
-            IssueDao issueDao = issueMainPageDtoList.get(i);
-            long id = issueDao.getId();
-            List<IssueLabelDto> issueLabelDtoList = new ArrayList<>();
-            IssueDto issueDto = new IssueDto(id, issueDao.getTitle(), issueDao.getContent(), issueDao.getUserName(),
-                    issueDao.getProfileUrl(), issueDao.getOpened(), issueDao.getCreatedAt(), issueDao.getClosedAt(),
-                    issueDao.getMilestoneName(), issueLabelDtoList);
-            Boolean flag = false;
-            while (i < issueMainPageDtoList.size() && id == issueDao.getId()) {
-                flag = true;
-                if (issueDao.getLabelId() != null) {
-                    issueLabelDtoList.add(
-                            new IssueLabelDto(issueDao.getLabelId(), issueDao.getLabelName(),
-                                    issueDao.getBackgroundColor(),
-                                    issueDao.getFontColor()));
-                }
-                i++;
-                if (i < issueMainPageDtoList.size()) {
-                    issueDao = issueMainPageDtoList.get(i);
-                }
-            }
-            issueDtoList.add(issueDto);
-            if (flag) {
-                i--;
-            }
-        }
-        issueListDto.setIssues(issueDtoList);
-        issueListDto.setUserList(fetchFilterUsers());
-        List<FilterLabelDto> filterLabelDtoList = fetchFilterLabels();
-        issueListDto.setLabelList(filterLabelDtoList);
-        List<FilterMilestoneDto> filterMilestoneDtoList = fetchFilterMilestones();
-        issueListDto.setMilestoneList(filterMilestoneDtoList);
+        List<IssueListPage> filteredIssueList = issueListMapper.findIssueList(filterDto);
 
-        issueListDto.setCountAllLabels(filterLabelDtoList.size());
-        issueListDto.setCountAllMilestones(filterMilestoneDtoList.size());
-        issueListDto.setCountOpenedIssues(issueDtoList.stream().filter(issueDto -> issueDto.getIsOpen()).count());
-        issueListDto.setCountClosedIssues(labelRepository.getTotalClosedIssueCount());
-        return issueListDto;
-    }
+        List<IssueListPage> issueMainPageDtoList = new ArrayList<>();
 
-    public List<FilterUserDto> fetchFilterUsers() {
-        return labelRepository.getFilterUserList();
-    }
+        for (IssueListPage issueListPage : filteredIssueList) {
+            issueListRepository.getIssues(issueListPage.getId()).forEach(e -> issueMainPageDtoList.add(e));
+        }
```

Comment:
iterator를 돌면서 forEach문을 사용하여 또 동작하네요 이러면 성능이슈가 생기지 않을까요?

Reply:
한번의 쿼리로 이슈목록을 조회했었는데 다중 조인에 동적쿼리까지 고려하니 너무 복잡도가 올라가서 쿼리를 한번 날려서 해당 페이지, 필터조건의 글 목록(10개)의 issue id만을 가져오고 issue id 10개에 대해서 서브 쿼리로 목록에 필요한 데이터를 조회하는 로직입니다.. forEach로 데이터를 담은 후에 다시 linkedHashmap에 넣으면서 조인으로 인한 중복 데이터를 제거하는 작업을 수행하는데요. 쿼리 횟수를 줄이면 복잡도가 높아져서 어느 선에서 절충해야할지 그 지점을 찾는게 어려운것 같습니다. 현재는 목록 한번 조회에 최대 11번의 쿼리가 발생하는 구조인데 이보다 적은 수의 쿼리로 데이터를 가져오도록 하는 것이 좋을지 궁금합니다.

Reply:
중복데이터가 조인으로 나온다는게 알기가 어렵네요. 실제로 쿼리 했을 때 데이터가 어떤식으로 출력되는지 알아야 중복데이터가 왜나오는지 이해할거 같네요.
만약 저라면 글 목록(issue id)를 가져온 후 issue id 리스트로 in 쿼리를 호출할 것 같아요. 그 이후 issue id와 그에 관계된 데이터를 Map으로 연결지을거 같아요.

Reply:
아 n+1 쿼리를 날리지말고 id 리스트를 가져온 후 추가 쿼리로 한번에 연관된 데이터를 가져오라는 말씀이신가요? 그 말씀이신거라면 충분히 이해했습니다..!

Reply:
```suggestion
        List<IssueListPage> issueMainPageDtoList = issueListMapper.findIssueList(filteredIssueIdList);
```
말씀하신대로 IN을 이용해서 총 2번의 쿼리로 가져오는 방식으로 수정했습니다. 이에따라 동적쿼리 작성이 필요해져서 mybatis로 매핑했습니다. 감사합니다!